### PR TITLE
Look for GPG agent socket also in $XDG_RUNTIME_DIR/gnupg

### PR DESCRIPTION
**Pull request by tomaszduda23**
It is implemented in similar way here https://github.com/maximbaz/yubikey-touch-detector/blob/ce2874a7a89dd7141195744a39bbd537ee243d24/detector/ssh.go#L22

### CODE REVIEW

File: gpgagent.go [hunk 1]
```diff
@@ -68,6 +70,16 @@ func NewConn() (*Conn, error) {
...
 		sockFile := path.Join(currentUser.HomeDir, ".gnupg", "S.gpg-agent")
+		IsSocket := func(path string) bool {
+			fileInfo, err := os.Stat(path)
+			if err != nil {
+				return false
+			}
+			return fileInfo.Mode().Type() == fs.ModeSocket
+		}
+		if runtimeDir, ok := os.LookupEnv("XDG_RUNTIME_DIR"); IsSocket(sockFile) == false && ok {
+			sockFile = path.Join(runtimeDir, "gnupg/S.gpg-agent")
```
Comment (felixfontein) [inline gpgagent.go]:
```suggestion
			sockFile = path.Join(runtimeDir, "gnupg", "S.gpg-agent")
```

**Comment by tomaszduda23** [comment]
@felixfontein any chance to get it merged?

**Comment by felixfontein** [comment]
@getsops/maintainers can someone else please take a look at this? I think this looks fine, but I don't know what impact `//go:build !appengine` has for example. Thanks.

**Comment by tomaszduda23** [comment]
`//go:build !appengine` was removed since it is not required.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@tomaszduda23 thanks for your contribution! And sorry it took so long...

**Pull request merged**